### PR TITLE
Add option to specify IP address.

### PR DESCRIPTION
**Pull request by bkolden**
Feature requires the **netifaces** package.

### CODE REVIEW

**Comment by npes87184** [comment]
Thank you 👍

So far so good. I just add some nitpicking requirement.

File: README.md [hunk 1]
```diff
@@ -56,6 +58,8 @@ optional arguments:
   --receive, -r         enable upload mode, received file will be stored at
                         given path.
   --port PORT, -p PORT  use custom port
+  --ip_addr {192.168.0.104,10.10.10.10}
+                        use a custom IP address
```
Comment (npes87184) [inline README.md]: I am concerned with the help "use a custom IP address", It is confused.
Maybe, something likes "specify an IP address"?

File: qr-filetransfer/qr-filetransfer [hunk 2]
```diff
@@ -236,6 +237,16 @@ def get_local_ip():
         sys.exit()
 
 
+def get_local_ips_available():
+    """Get a list of all local IPv4 addresses except localhost"""
+    ips = []
+    for iface in netifaces.interfaces():
+        ips.extend([x["addr"] for x in netifaces.ifaddresses(iface).get(netifaces.AF_INET, []) if x])
```
Comment (npes87184) [inline qr-filetransfer/qr-filetransfer]: In addition to check the existence of `x`, please also check the existence of the key `addr`.
e.g. `ips.extend([x["addr"] for x in netifaces.ifaddresses(iface).get(netifaces.AF_INET, []) if x and "addr" in x])`

Thanks

File: README.md (hunk 1 above)
Comment (bkolden) [reply README.md]: I used the phrase `custom IP Address` to match the same wording as the already existing  `custom port` phrase.  I can certainly change it as requested.

Comment (sdushantha) [reply README.md]: > I used the phrase custom IP Address to match the same wording as the already existing custom port phrase.

@bkolden  I understand what you mean. A _custom port_ can be a made up port, but a _custom IP address_ can not be a made up IP address. 

So think `specify an IP address` would be better :)

**Comment by sdushantha** [comment]
Thanks for the PR :)

Can I ask, in what type of scenario would you use this flag?

**Comment by bkolden** [comment]
I use OpenVPN for work, which has a default route through the VPN network via the TUN device that the OpenVPN client creates.  The IP address that the existing *qr-filetransfer* code uses is the one bound to that TUN device, not the local Ethernet or WiFi device.  This results in the wrong IP address being encoded in the QR code.  Specifying the desired IP resolves that issue.

**Comment by sdushantha** [comment]
@bkolden  Good thinking :+1:

**Pull request merged**